adapters/repos/db/shard_dimension_tracking.go

```diff
@@ -0,0 +1,51 @@
+package db
+
+import (
+	"encoding/binary"
+	"time"
+
+	"github.com/semi-technologies/weaviate/adapters/repos/db/helpers"
+)
+
+func (s *Shard) Dimensions() int {
+	b := s.store.Bucket(helpers.DimensionsBucketLSM)
+	if b == nil {
+		return 0
+	}
+
+	c := b.MapCursor()
+	sum := 0
+	for k, v := c.First(); k != nil; k, v = c.Next() {
+		dimLength := binary.LittleEndian.Uint32(k)
+		sum += int(dimLength) * len(v)
+	}
+	c.Close()
+
+	return sum
+}
+
+func (s *Shard) initDimensionTracking() {
+	if !s.index.Config.TrackVectorDimensions {
+		return
+	}
+
+	go func() {
+		t := time.NewTicker(5 * time.Second) // 5 minutes
```

Comment:
Comment says 5 minutes but ticker is set to 5 seconds, which is true?

Reply:
I think this comment was originally from me. The motivation was that it happened super frequently during the development (5s), but once it's done, we can have it happen less often (5min) as there is some cost to this. (The cost is rather low due to the inverted index approach, but it's not free).